packages/kbn-openapi-generator/docs/USAGE_GUIDE.md

```diff
@@ -0,0 +1,432 @@
+
+### How to organize files with OpenAPI specs?
+
+You have quite a lot of freedom on organizing your OpenAPI specs and code generator supports any valid OpenAPI. Though we recommend to follow rules derived from using OpenAPI specs and code generation in Rule Management team
+
+- All your OpenAPI specs should be under a single domain folder with clear ownership assigned to your team via `.CODEOWNERS` file.
+- Prefer placing OpenAPI specs in a package (see `kbn-securitysolution-lists-common` or `kbn-securitysolution-exceptions-common` for examples). Generated artifacts can be easily imported into the other packages or plugins. Having specs in a package simplify reusing (you don't need long relative paths `../../../../my.schema.yaml`) of common OpenAPI primitives like `NonEmptyString` or `UUID` defined in `kbn-openapi-common` package.
+- Split API endpoint declarations into separate files. The rule of thumb here is having one HTTP Method declaration per file. For example you want to declare two endpoints `GET /api/my/data` and `POST /api/my/data` so you need to create two OpenAPI spec files for them.
+- Make sure you define `operationId` for each path item.
+- Make sure file name matches defined `operationId`.
+- Feel free to group OpenAPI specs inside domain folder in smaller groups according business logic.
+- In the majority of cases importing `.gen.ts` file directly may complicate further OpenAPI schemas reorganization. To mitigate this problem prefer usage of index files.
+
+> **Info:** By default OpenAPI code generator expects OpenAPI spec files named `<my-name>.schema.yaml`. If it's absolutely necessary to use another suffix make sure OpenAPI code generator picks file up by expecting generating script where `generate()` function is invoked.
+
+#### Example
+
+You can find an example package's folder structure
+
+- kbn-solutionname-domain-name-common
+  - api
+    - business feature A
+      - create_entity.schema.yaml
+      - delete_entity.schema.yaml
+        ...
+    - business feature B
+      - read_configuration.schema.yaml
+      - update_featureB_entity.schema.yaml
+        ...
+  - scripts - openapi_generate.js
+    ...
+
+### How to enable code generation?
+
+By default OpenAPI code generator generation produces artifacts for `components` defined in OpenAPI specs but disabled for `paths`. `x-codegen-enabled` custom property allows to enable/disable code generation for particular path item or components. `x-codegen-enabled: true` should be always added to path items you want to enable code generation for.
+
+> **Warning:** When code generation for path items is enabled OpenAPI code generator produces Zod schemas and TS types for request body, request parameters and successful response. ==You DON'T need to define schemas for request body, request parameters and successful response manually.== If you don't see such artifacts most probably path items code generation is disabled.
```

Comment:
Looks like highlighting with `==` doesn't work if it's within a blockquote. But doing `*` or `**` works fine.